### ISSUE TITLE
Analytics not working

### ISSUE THREAD

**Issue by Murph22** [Estimote/Android-SDK#154]
When trying to set up analytics, we get the following error:

```
com.estimote.sdk.cloud.internal.AnalyticsManager$2.failure:253

 Unable to obtain owned devices for Analytics
                                                                   com.estimote.sdk.exception.EstimoteServerException: InternalEstimoteApi.getBeacons: HTTP method annotation is required (e.g., @GET, @POST, etc.).
                                                                       at com.estimote.sdk.cloud.internal.InternalEstimoteCloud$6.failure(InternalEstimoteCloud.java:347)
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.CallbackRunnable$2.run(CallbackRunnable.java:53)
                                                                       at android.os.Handler.handleCallback(Handler.java:739)
                                                                       at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                       at android.os.Looper.loop(Looper.java:158)
                                                                       at android.app.ActivityThread.main(ActivityThread.java:7229)
                                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
                                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
                                                                    Caused by: com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RetrofitError: InternalEstimoteApi.getBeacons: HTTP method annotation is required (e.g., @GET, @POST, etc.).
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:399)
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RestAdapter$RestHandler.access$000(RestAdapter.java:219)
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RestAdapter$RestHandler$1.obtainResponse(RestAdapter.java:277)
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.CallbackRunnable.run(CallbackRunnable.java:42)
                                                                       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                                                                       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.Platform$Android$2$1.run(Platform.java:134)
                                                                       at java.lang.Thread.run(Thread.java:818)
                                                                    Caused by: java.lang.IllegalArgumentException: InternalEstimoteApi.getBeacons: HTTP method annotation is required (e.g., @GET, @POST, etc.).
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RestMethodInfo.methodError(RestMethodInfo.java:106)
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RestMethodInfo.parseMethodAnnotations(RestMethodInfo.java:178)
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RestMethodInfo.init(RestMethodInfo.java:116)
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:293)
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RestAdapter$RestHandler.access$000(RestAdapter.java:219) 
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.RestAdapter$RestHandler$1.obtainResponse(RestAdapter.java:277) 
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.CallbackRunnable.run(CallbackRunnable.java:42) 
                                                                       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
                                                                       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
                                                                       at com.estimote.sdk.repackaged.retrofit_v1_9_0.retrofit.Platform$Android$2$1.run(Platform.java:134) 
                                                                       at java.lang.Thread.run(Thread.java:818)
```

We initialize the SDK in our application file as follows: 

```
EstimoteSDK.initialize(this, "APP-ID", "APP-TOKEN");       
EstimoteSDK.enableMonitoringAnalytics(true);
  EstimoteSDK.enableRangingAnalytics(true);

```

**Comment by pawelDylag** [comment]
Hello @Murph22 !

Thanks for reporting that! This kind of error occurs, when annotations are stripped by ProGuard. You can try disabling it and check if that helps. If you really need to use it, you can try adding this to your proguard-rules file:
```
-keepattributes *Annotation*
```
Here is a discussion with more info: https://github.com/square/retrofit/issues/435
Tell me if that helps and have a nice day :)

**Comment by Murph22** [comment]
Hey @pawelDylag,

Thanks for the quick response. Since we were already using retrofit in our app we already had that line in our proguard-rules file as follows: 
```
-keepattributes SourceFile,LineNumberTable,*Annotation*,Signature
```

Is there any other fix you can think of that can resolve this issue?

**Comment by Luxilia** [comment]
I'm also having problems with EstimoteCloud and retrofit, and I have proguard disabled.
Get "No retrofit annotation for parameter #1" and "Only one http request allowed, found GET and GET" when trying to use the cloud service.

**Comment by Arun-intelipower** [comment]
Hi I'm getting the same Error my stack trace is as follows

**Comment by pawelDylag** [comment]
Hello @Arun-intelipower, @Murph22 

We recently released version 1.0.0 of our SDK ([See migration guide](https://github.com/Estimote/Android-SDK/blob/master/Docs/switching_to_1.0.0.md)). Could you confirm if this problem still occurs for you? If so, we will investigate it further and add some fixups :)
